packages/twenty-front/src/modules/object-record/record-index/components/RecordIndexTableContainerEffect.tsx

```diff
@@ -68,5 +74,37 @@ export const RecordIndexTableContainerEffect = () => {
     );
   }, [setRecordCountInCurrentView, setOnEntityCountChange]);
 
+  const setViewFieldsAggregateOperations = useRecoilCallback(
+    ({ set, snapshot }) =>
+      (viewField: ViewField) => {
+        const aggregateOperationForViewField = snapshot
+          .getLoadable(
+            aggregateOperationForViewFieldState({
+              viewFieldId: viewField.id,
+            }),
+          )
+          .getValue();
+
+        if (aggregateOperationForViewField !== viewField.aggregateOperation) {
+          set(
+            aggregateOperationForViewFieldState({
+              viewFieldId: viewField.id,
+            }),
+            viewField.aggregateOperation,
+          );
+        }
+      },
+    [],
+  );
+
+  useEffect(() => {
+    for (const viewField of (currentViewWithSavedFiltersAndSorts?.viewFields || [])) {
+      setViewFieldsAggregateOperations(viewField);
+    };
+  }, [
+    currentViewWithSavedFiltersAndSorts?.viewFields,
+    setViewFieldsAggregateOperations,
+  ]);
+
```

Comment:
Or 

currentViewWithSavedFiltersAndSorts?.viewFields.forEach(
      setViewFieldsAggregateOperations,
    );

    Works in that case, why not ?

Reply:
Here we can see why having the naming in singular fits with the fact that we apply a single operation on each item, it's coherent if we read it : `For each view field, set aggregate operation`